### ISSUE TITLE
[a11y][Violation][1.1.1 Non-text Content] Multiple SVG elements has no accessible names

### ISSUE THREAD

**Issue by nkn2022** [jupyterlab/jupyterlab#17460]
## Description

I am running an accessibility test suite with Jupyter using the **IBM Equal Access Accessibility Checker** tool on Chrome browser. Few issues found on the `data-icon="ui-components:XX` are missing to reach a first formal level of compliance. So kindly help look into these. 

Issues reported were violating **WCAG 2.2** requirements. 
Ref: https://www.ibm.com/able/requirements/checker-rule-sets

Listing of violations categorized based on the checkpoint. 

****Violation of Rule:** `svg_graphics_labelled`**

> List of issues with elements that violate this rule
> > **Issue 1** 
> > The SVG element has no accessible name
> > **Sample code references(not showing all):** 
> > `<svg class="" data-icon="ui-components:save" width="32" viewBox="0 0 32 32" height="32" xmlns="http://www.w3.org/2000/svg">`
> > `<svg data-icon="ui-components:caret-down-empty" viewBox="0 0 18 18" width="16" xmlns="http://www.w3.org/2000/svg">`
> > `<svg data-icon-id="02358d44-6b12-4c0a-96fc-3c3c8147d4c4" data-icon="ui-components:ellipses" viewBox="0 0 24 24" width="16" xmlns="http://www.w3.org/2000/svg">`
...

Possible Fixes: 

> What to do
> Ensure the non-decorative SVG element has an accessible name that is not empty:
> 
> Add an aria-labelledby attribute to the element that points to visible text on the page that is a meaningful label.
> Or, add an aria-label attribute to the element.
> Or, add a direct child <title> element.
> Or, add an xlink:title attribute on a link.
> Or, for text container elements, add the text content.
> Or, only if the design cannot have a visible label, use the title attribute to provide a label.

## Reproduce

- Install Chrome extension for IBM accessibility checker tool. Ref: https://www.ibm.com/able/toolkit/verify/automated
- Open JupyterLab UI on Chrome browser. Open `notebook` file to edit.
- Run the tool and we can see the above `Violation`s reported.

## Expected behavior

No violations are reported on the scanning to reach a first formal level of compliance.

## Context

- Operating System and version: MacOS Sonoma Version 14.7.3
- Browser and version: Chrome Version 134.+
- JupyterLab version: Version 4.3.6

**Comment by krassowski** [comment]
`dsx` icons are not jupyterlab icons I think? In general, if these are part of a button then we should either:
- mark them as decorative (for `img` this is `alt=""` as per https://www.w3.org/WAI/tutorials/images/decorative/, not sure what is equivalent for SVGs) if there is a label next to it, or otherwise
- add what would be displayed as a label in `aria-label` or `title`

**Comment by nkn2022** [comment]
@krassowski This is the icon that shows up for the element `dsx:icon_add`. Is this not part of jupyterlab icons? I could not find any `dsx` reference when searching the code. So not completely sure. 

<img width="1328" alt="Image" src="https://github.com/user-attachments/assets/e0047cd5-9d8c-4085-a231-a6c0db6a339c" />

**Comment by nkn2022** [comment]
To mark SVGs as decorative we should add `aria-hidden="true"`  here.

**Comment by nkn2022** [comment]
@krassowski I found that the icons with `dsx:icon_` tags are coming from overrides on our side. So, you are right, it is not part of the jupyter labs icons. I will update the description to remove this.

**Comment by nkn2022** [comment]
@krassowski I have proposed a fix for this issue. Once this is issue is triaged and marked as an issue, you may review: https://github.com/jupyterlab/jupyterlab/pull/17472
With this change all icons that do not have a `title` defined will add this property and considered decorative.
